src/integrationtests/java/com/aws/greengrass/integrationtests/extensions/BridgeIntegrationTestExtension.java

```diff
@@ -79,16 +92,96 @@ public class BridgeIntegrationTestExtension implements AfterTestExecutionCallbac
     MQTTClientKeyStore clientKeyStore = new InitOnceMqttClientKeyStore();
 
     @Override
-    public void interceptTestMethod(Invocation<Void> invocation, ReflectiveInvocationContext<Method> invocationContext,
-                                    ExtensionContext extensionContext) throws Throwable {
-        initializeContext(extensionContext);
-        String configFile = getConfigFile(extensionContext);
-        if (configFile == null) {
-            invocation.proceed();
-            return;
+    public boolean supportsTestTemplate(ExtensionContext context) {
+        return context.getTestMethod().isPresent()
+                && AnnotationUtils.isAnnotated(context.getTestMethod().get(), BridgeIntegrationTest.class);
+    }
+
+    @Override
+    public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(ExtensionContext context) {
+        BridgeIntegrationTest annotation = context.getRequiredTestMethod().getAnnotation(BridgeIntegrationTest.class);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;context&#39; (index: 0 | type: ExtensionContext). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other methods validating the parameter: &#39;supportsTestTemplate&#39;. The same parameter type is validated here for example: aws-greengrass-mqtt-bridge/src/integrationtests/java/com/aws/greengrass/integrationtests/extensions/BridgeIntegrationTestExtension.java:96. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).